addendumer.go

```diff
@@ -12,3 +17,149 @@ type Addendumer interface {
 	String() string
 	Validate() error
 }
+
+type addenda struct {
+	ID string `json:"id"` // EntryDetail.ID that this addenda belongs to
+
+	Addenda Addendumer
+}
+
+// UnmarshalJSON attempts to parse data as each AddendaXX record.
+func (a *addenda) UnmarshalJSON(data []byte) error {
+	addenda02 := NewAddenda02()
+	if json.Unmarshal(data, &addenda02) == nil && addenda02.Validate() == nil {
+		a.ID = addenda02.ID
+		a.Addenda = addenda02
+		return nil
+	}
+
+	addenda05 := NewAddenda05()
+	if json.Unmarshal(data, &addenda05) == nil && addenda05.Validate() == nil {
+		a.ID = addenda05.ID
+		a.Addenda = addenda05
+		return nil
+	}
+
+	addenda10 := NewAddenda10()
+	if json.Unmarshal(data, &addenda10) == nil && addenda10.Validate() == nil {
+		a.ID = addenda10.ID
+		a.Addenda = addenda10
+		return nil
+	}
+
+	addenda12 := NewAddenda12()
+	if json.Unmarshal(data, &addenda12) == nil && addenda12.Validate() == nil {
+		a.ID = addenda12.ID
+		a.Addenda = addenda12
+		return nil
+	}
+
+	addenda13 := NewAddenda13()
+	if json.Unmarshal(data, &addenda13) == nil && addenda13.Validate() == nil {
+		a.ID = addenda13.ID
+		a.Addenda = addenda13
+		return nil
+	}
+
+	addenda14 := NewAddenda14()
+	if json.Unmarshal(data, &addenda14) == nil && addenda14.Validate() == nil {
+		a.ID = addenda14.ID
+		a.Addenda = addenda14
+		return nil
+	}
+
+	addenda15 := NewAddenda15()
+	if json.Unmarshal(data, &addenda15) == nil && addenda15.Validate() == nil {
+		a.ID = addenda15.ID
+		a.Addenda = addenda15
+		return nil
+	}
+
+	addenda16 := NewAddenda16()
+	if json.Unmarshal(data, &addenda16) == nil && addenda16.Validate() == nil {
+		a.ID = addenda16.ID
+		a.Addenda = addenda16
+		return nil
+	}
+
+	addenda17 := NewAddenda17()
+	if json.Unmarshal(data, &addenda17) == nil && addenda17.Validate() == nil {
+		a.ID = addenda17.ID
+		a.Addenda = addenda17
+		return nil
+	}
+
+	addenda18 := NewAddenda18()
+	if json.Unmarshal(data, &addenda18) == nil && addenda18.Validate() == nil {
+		a.ID = addenda18.ID
+		a.Addenda = addenda18
+		return nil
+	}
+
+	addenda98 := NewAddenda98()
+	if json.Unmarshal(data, &addenda98) == nil && addenda98.Validate() == nil {
+		a.ID = addenda98.ID
+		a.Addenda = addenda98
+		return nil
+	}
+
+	addenda99 := NewAddenda99()
+	if json.Unmarshal(data, &addenda99) == nil && addenda99.Validate() == nil {
+		a.ID = addenda99.ID
+		a.Addenda = addenda99
+		return nil
+	}
+
+	return nil
+}
+
+type addendas []addenda
+
+// addToFile will iterate through the File's EntryDetail records and add cooresponding
+// Addendumer records. This is designed to only be used by addAddendasFromJSON.
+func (a addendas) addToFile(f *File) {
+	for i := range a {
+		addenda := a[i]
+		for j := range f.Batches {
+			entries := f.Batches[j].GetEntries()
+			for k := range entries {
+				if addenda.ID == entries[k].ID {
```

Comment:
This logic forces Addenda ID's to match their EntryDetail ID. Is that ok? Can an EntryDetail with multiple Addneda records have each Addenda with the same ID?

Reply:
Wouldn't we want a unique ID for each Addenda Record with a FK to the entryDetailID?

Reply:
Right, the SequenceNumber is i order and ascending (starting with 1) for each addenda for an entry detail, and entryDetailSequenceNumber is the same as the last seven digits of the trace number of the related Entry Detail Record.

So what makes each addenda unique is the combination of entryDetailSequenceNumber and Sequence Number.  That seems ok then if ID is the same

  
I guess I wasn't sure what ID represented

Reply:
I have a different idea to try.